### PR TITLE
feat: add product list page

### PR DESCRIPTION
**Pull request by cmgriffing-at-gitkraken**

### CODE REVIEW

**Comment by cmgriffing-at-gitkraken** [comment]
Hey @cmgriffing-at-gitkraken, while reviewing your PR, I'd suggest the following code changes:

👉 [**Update color pallette**](https://gitkraken.dev/drafts/8a8eb425-bc0a-4204-90ce-72cb2ca416b6?source=PRComment)
>

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/8a8eb425-bc0a-4204-90ce-72cb2ca416b6?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPTF9yNmNzNXdadjNJIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

**Comment by cmgriffing-at-gitkraken** [comment]
Hey @cmgriffing-at-gitkraken, while reviewing your PR, I'd suggest the following code changes:

👉 [**Use Products API instead of Widgets and Gadgets separately**](https://gitkraken.dev/drafts/0412a317-a05f-4922-b341-ee0c6994f30b?source=PRComment)
>

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/0412a317-a05f-4922-b341-ee0c6994f30b?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPTF9yNmNzNXdadjNJIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

**Comment by cmgriffing-at-gitkraken** [comment]
Hey @cmgriffing-at-gitkraken, while reviewing your PR, I'd suggest the following code changes:

👉 [**Reword page title and description**](https://gitkraken.dev/drafts/3ee03299-a4a1-4f87-87fd-089919faea33?source=PRComment)
>https://github.com/cmgriffing-at-gitkraken/enterprise-todo-app/pull/3

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/3ee03299-a4a1-4f87-87fd-089919faea33?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPTF9yNmNzNXdadjNJIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

File: src/types.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export interface Gadget {
+  id: string;
+  name: string;
+  description: string;
+}
+
+export interface GadgetsResponse {
+    gadgets: Gadget[];
+  }
```
Comment (cmgriffing-at-gitkraken-backend) [inline src/types.ts:20]: The rest of this looks great, but it seems the indentation is a little funky here.

We should probably create a task to add a prettier auto formatting commit hook. Not needed in this PR, but something we want to track.

Comment (cmgriffing-at-gitkraken) [reply src/types.ts]: Great catch. Fixed.

**Pull request merged**